src/main.cpp

```diff
@@ -1521,14 +1527,14 @@ bool CTransaction::CheckInputs(CValidationState &state, CCoinsViewCache &inputs,
             if (!GetCoinAge(state, inputs, nCoinAge))
                 return error("CheckInputs() : %s unable to get coin age for coinstake", GetHash().ToString().c_str());
             int64 nStakeReward = GetValueOut() - nValueIn;
-            if (nStakeReward > GetProofOfStakeReward(nCoinAge) - GetMinFee() + MIN_TX_FEE)
+            if (nStakeReward > GetProofOfStakeReward(nCoinAge) - GetMinFee() + MIN_TX_FEE*10)
```

Comment:
I think this will allow bigger rewards after the protocol change.

In the current protocol when the CoinStake has a size that requires the minimum fee (0.01 PPC), it is allowed to get the full reward (proof reward - 0.01 + 0.01).

With this PR and after the protocol switch, when the CoinStake has a size that requires the minimum fee (0.001 PPC), it is allowed to get the full reward + 0.009 PPC (proof reward - 0.001 + 0.01).

Reply:
First of all, any protocol change must be protected by a protocol switch like IsProtocolV07() etc. This change appears unprotected, even though older blocks are ok as it is a loosening of the check, it's not okay during the upgrade window and is a potential cause for forks.

This also appears to be an unnecessary protocol change to me. Generally speaking minor protocol changes should be discouraged as there is a cost to maintain any protocol change. It's not just a matter of how many lines of code to maintain and its associated merge cost but also who could remember all the little protocol changes accurately.

Reply:
It is a confusion caused by changing MIN_TX_FEE, see comment at the definition of MIN_TX_FEE.